include/envoy/http/filter.h

```diff
@@ -224,6 +224,12 @@ class StreamFilterBase {
   virtual ~StreamFilterBase() {}
 
   /**
+   * This routine is called on filter creation, allowing filters to subscribe to watermark events
+   * on the downstream stream and downstream connection.
+   */
+  virtual void provideWatermarkCallbacks(Http::DownstreamWatermarkProvider& provider) PURE;
```

Comment:
Yep.  I suspect we're often going to have many filters and I wanted to avoid 1:M*N where M is the number of streams and N is the number of filters.
Adding detail to the flow control doc.